pkg/services/sqlstore/searchstore/builder.go

```diff
@@ -37,7 +37,7 @@ func (b *Builder) ToSQL(limit, page int64) (string, []any) {
 		INNER JOIN dashboard ON ids.id = dashboard.id`)
 	b.sql.WriteString("\n")
 
-	if b.Features.IsEnabledGlobally(featuremgmt.FlagNestedFolders) {
+	if b.Features.IsEnabledGlobally(featuremgmt.FlagNestedFolders) && !b.Features.IsEnabledGlobally(featuremgmt.FlagKubernetesFoldersServiceV2) {
```

Comment:
What's the reasoning for the feature toggle check here?

Reply:
I did that so we could experiment with running Grafana without depending on the `folder` table.

You are right that this is not actually needed, since we will move away from `searchstore` and start using the indexing service to search for dashboards

What do you prefer:
- remove the feature toggle check from here?
- keep it here but add a `TODO` comment explaining that this will change soon?

Reply:
I'd prefer to remove it from here, so that we keep the legacy functionality untouched & can switch over to using the indexer

Reply:
Agree with you and I am going to revert this change now